File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int XOR1=0;
        for(int i=0;i<=nums.length;i++){
            XOR1 = XOR1^i;

        }
        int XOR2=0;
        for(int i=0;i<nums.length;i++){
            XOR2 = XOR2^nums[i];
        }
        return XOR1^XOR2;
        
    }
}

File: 268-missing-number/missing-number.c
int missingNumber(int* nums, int numsSize) {
    int sum =0;
    int actualsum =0;
    for(int i=0;i<=numsSize;i++){
        sum+=i;
    }
    for(int i=0;i<numsSize;i++){
        actualsum+=nums[i];
    }
    return sum-actualsum;

    
}